lib/lineage.js

```diff
@@ -35,7 +35,8 @@ const buildHydratedDoc = (doc, lineage) => {
     // if the parent doc isn't found the element in lineage is null,
     // so retain the stub.
     const parent = lineage.shift() || currentStub;
-    if (!parent) {
+    if (_.isEmpty(parent)) {
+      delete currentParent.parent;
```

Comment:
I feel like this line fits better in the `minify` function. This is particularly true in the other branches where the shared library minify is called when creating the records too.